### DISCUSSION TITLE
How do I get combinatorial to choose from the first to the last in a wild cards file?

### DISCUSSION THREAD

**Discussion by bobpuffer1** [adieyal/sd-dynamic-prompts#521] [Q&A]
My wildcards file contains four value lines. The initial line chosen to start off the cycle through the whole file is random when combinatorial is selected. I'd like to have it choose the first value, then the second... and do on but its initial pick is always random. This was not the behavior when the values were placed in an inline dynamic phrase, i.e., {pick1|pick2|pick3}.

**Comment by andupotorac** [comment]
I'd also like to know how I can tell SD to go through each line in my wildcards when using Dynamic Prompts.

**Reply by bobpuffer1** [reply]
Combinatorial will do that. It'll walk through all the permutations of each wild card starting from the last to the first in the prompt. My problem is it doesn't start at the beginning of the wild cards file. It starts somewhere in the middle. It'll walk through all of them, but I wanted to start at the very beginning.

**Reply by adieyal** [reply]
Can you check if it selects them in alphabetical order? (i.e not random)

**Reply by bobpuffer1** [reply]
Yep. You're right takes it in alphabetical order. For me, that would not be the desired behavior.

**Reply by bobpuffer1** [reply]
Desired behavior would be that it would choose randomly from the file, because what happens is you run a generation of 32 images on combinatorial, and it chooses your last dynamic prompt, and goes to all of the iterations for that alphabetically, and as it does that it goes through the iterations of your second to the last dynamic prompt. And because it goes through those alphabetically anytime you have combinatorial on you start over with each new generation, it produces exactly the same stuff. That's a problem. I'm going to write up an issue.
